source/CMakeLists.txt

```diff
@@ -127,6 +130,14 @@ foreach(build ${DEAL_II_BUILD_TYPES})
     RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${DEAL_II_EXECUTABLE_RELDIR}"
     )
 
+  separate_arguments(_compile_options UNIX_COMMAND
+    "${DEAL_II_CXX_FLAGS} ${DEAL_II_CXX_FLAGS_${build_uppercase}}"
+    )
+  shell_escape_option_groups(_compile_options)
+  target_compile_options(${DEAL_II_NAMESPACE}_${build_lowercase} PUBLIC
+    $<$<COMPILE_LANGUAGE:CXX>:${_compile_options}>
+    )
```

Comment:
This is a big issue:

As neat as it seems we have to be very careful with adding `PUBLIC` compile and link options to the targets. 
Because this cannot be overriden by user code any more.

What I mean is this:

Suppose deal.II had been configured with `-std=c++17` (and the compiler is defaulting to `-std=c++20`. Then neither
```
set(CMAKE_CXX_FLAGS "-std=c++20")
add_executable(${TARGET} ${TARGET_SRC})
target_link_libraries(${TARGET} dealii::dealii)
```
nor
```
add_executable(${TARGET} ${TARGET_SRC})
set_property(TARGET ${TARGET} PROPERTY COMPILE_OPTIONS "-std=c++20")
target_link_libraries(${TARGET} dealii::dealii)
```
will allow you to override this compile flag any more. CMake will always order the compile options that come with `dealii::dealii_(debug|release)` last.


Remark: Funny enough, `set_property(TARGET ${TARGET} PROPERTY CXX_STANDARD 20)` works because in this case the language switch will be applied last... but this isnot the point I want to make here. Think for example about packaging for Debian where we are forced to have a `-march=generic` but suppose you want to use `-march=native` for your own target. This will not work.

The bottom line is that I think that the `dealii::dealii_debug` and `dealii::dealii_release` target will have to be free from any compile and link options.

So what about adding these target properties to the `dealii::dealii` dummy target instead? (with the same `$<CONFIG:...>` generator expression logic).